inc/class-s3-media-sync.php

```diff
@@ -34,10 +38,16 @@ public function get_settings_handler() {
 	}
 
 	public function get_s3_bucket() {
+		if ( $this->bucket === null ) {
+			return '';
+		}
 		return $this->bucket->get_name();
 	}
 
 	public function get_s3_bucket_url() {
+		if ( $this->bucket === null ) {
```

Comment:
Same as `get_s3_bucket()`: returning `''` as a sentinel is brittle. Prefer `null` and let callers decide how to display/format. Suggested change:

```php
public function get_s3_bucket_url() {
    return $this->bucket ? 's3://' . $this->bucket->get_name() : null;
}
```

Update downstream callers and docblocks to handle `?string`.